react/features/base/tracks/actions.web.ts

```diff
@@ -317,3 +292,225 @@ export function openAllowToggleCameraDialog(onAllow: Function, initiatorId: stri
         initiatorId
     });
 }
+
+/**
+ * Sets the GUM pending state for the tracks that have failed.
+ *
+ * NOTE: Some of the track that we will be setting to GUM pending state NONE may not have failed but they may have
+ * been requested. This won't be a problem because their current GUM pending state will be NONE anyway.
+ *
+ * @param {JitsiLocalTrack} tracks - The tracks that have been created.
+ * @param {Function} dispatch - The redux dispatch function.
+ * @returns {void}
+ */
+export function setGUMPendingStateOnFailedTracks(tracks: Array<any>, dispatch: IStore['dispatch']) {
+    const tracksTypes = tracks.map(track => {
+        if (track.getVideoType() === VIDEO_TYPE.DESKTOP) {
+            return MEDIA_TYPE.SCREENSHARE;
+        }
+
+        return track.getType();
+    });
+    const nonPendingTracks = [ MEDIA_TYPE.AUDIO, MEDIA_TYPE.VIDEO ].filter(type => !tracksTypes.includes(type));
+
+    dispatch(gumPending(nonPendingTracks, IGUMPendingState.NONE));
+}
+
+/**
+ * Creates and adds to the conference the initial audio/video tracks.
+ *
+ * @param {Array<MediaType>} devices - Array with devices (audio/video) that will be used.
+ * @returns {Function}
+ */
+export function createAndAddInitialAVTracks(devices: Array<MediaType>) {
+    return (dispatch: IStore['dispatch']) => {
+        dispatch(gumPending(devices, IGUMPendingState.PENDING_UNMUTE));
+
+        return dispatch(createInitialAVTracks({ devices }))
+        .then(({ tracks, errors }) => {
+            setGUMPendingStateOnFailedTracks(tracks, dispatch);
+            dispatch(displayErrorsForCreateInitialLocalTracks(errors));
+
+            return Promise.allSettled(tracks.map((track: any) => {
```

Comment:
I've included some async/await syntax. Note that I haven't removed the Promise.allSettled because we want to execute the useVideo/AudioStream simultaneously. Also removed the return because we actually don't use the result for now.

Why do we want to check if the promise is fulfilled or rejected? Do you mean the `Promise.allSettled`? I guess it will be nice to show an error notification or something but currently even on unmutes we don't handle the error at all or maybe there is a common logic in a middleware or something. So we can do this as part of an improvement PR for everything that adds or replaces tracks. Do you have any other reasons to want to check for rejected/fullfiled?